apps/src/util/KeyHandler.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * A KeyHandler class that can listen for events with multiple keys pressed.
+ */
+class KeyHandler {
+  constructor(element) {
+    this.pressedKeyMap = {};
+    this.keyHandlers = [];
+
+    element.addEventListener('keydown', e => this.onKeyDown_(e.key));
+    element.addEventListener('keyup', e => this.onKeyUp_(e.key));
+  }
+
+  /**
+   * Register an event to be called when a set of keys are pressed.
+   *
+   * @param {String[]} keys the set of keys that trigger this event.
+   * @param {Function} callback function to call when the keys are pressed.
+   */
+  registerEvent(keys, callback) {
+    this.keyHandlers.push([keys, callback]);
+  }
+
+  onKeyDown_(key) {
+    // Ignore keys that we don't have mappings for
+    const keysToCheck = this.keyHandlers.map(([keys]) => keys).flat();
```

Comment:
Nit: could this be `[keys, _]` (or not sure if there's some other convention for noting unused arguments in JS) for the map callback arg to just emphasize the structure of each element in `keyHandlers`? Relatedly, no TS here? 😝

Reply:
Oh yeah, I wasn't sure if that was just our convention in Ruby or JS as well? I don't have strong feelings either way, other than I've seen this kind of syntax when destructuring object properties. But happy to use `_` if that's clearer!

Also haha re: no TS - I figured given the engineering team discussion that "shared" code (this being in /util) should stay JS (though I wouldn't be opposed to converting this class)

Reply:
eh skimming through our code, seems like more of a Ruby thing. Fine to leave as-is...curious if @thomasoniii has thoughts on a best practice here. Also, got it re: shared code/TS, makes sense!